### PR TITLE
[web] Improve dom canvas speed. Fix density calculation regression in recomputeTransformClip

### PR DESCRIPTION
**Pull request by ferhatb**
## Description

- Dom canvas now uses flt-picture instead of creating extra dom-canvas node.
- computeDensity now uses isIdentityOrTranslation() to fix regression in update_many_child_layers benchmark.

```
═════════════════════════╡ ••• Final A/B results ••• ╞══════════════════════════
Score	Average A (noise)	Average B (noise)	Speed-up
bench_update_many_child_layers.html.drawFrameDuration.average	8641.60 (1.38%)	2418.36 (2.71%)	3.57x	
bench_update_many_child_layers.html.drawFrameDuration.outlierRatio	1.80 (5.63%)	1.10 (1.48%)	1.64x
bench_update_many_child_layers.html.totalUiFrame.average	28969.70 (0.66%)	9189.70 (1.25%)	3.15x
```

- Micro optimizations around global final domRenderer that does extra lazyinit lookup.
- Micro optimizations around style setting:

```
Before:
    apply$0: function() {
      var t1 = this.rootElement.style,
        t2 = "translate(" + H.S(this.dx) + "px, " + H.S(this.dy) + "px)";
      t1.toString;
      C.CssStyleDeclaration_methods._setPropertyHelper$3(t1, C.CssStyleDeclaration_methods._browserPropertyName$1(t1, "transform"), t2, "");
    },
After:
 apply$0: function() {
      var t2,
        t1 = this.rootElement;
      t1.toString;
      t2 = "translate(" + H.S(this.dx) + "px, " + H.S(this.dy) + "px)";
      t1.style.transform = t2;
    },
```

## Related Issues

https://github.com/flutter/flutter/issues/69910
https://github.com/flutter/flutter/issues/59404

## Tests

Covered by web_benchmarks.dart.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [x] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

File: lib/web_ui/lib/src/engine/vector_math.dart [hunk 1]
```diff
@@ -432,6 +432,27 @@ class Matrix4 {
       _m4storage[14] == 0.0 &&
       _m4storage[15] == 1.0;
 
+  /// Whether transform is identity or simple translation using m[12,13,14].
+  ///
+  /// We check for [15] first since that will eliminate bounds checks for rest.
+  bool isIdentityOrTranslation() =>
+      _m4storage[15] == 1.0 &&
+      _m4storage[0] == 1.0 // col 1
+          &&
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/vector_math.dart:441]: This doesn't look right. Maybe move `&&` before `// col 1`?

File: lib/web_ui/lib/src/engine/vector_math.dart [hunk 2]
```diff
@@ -432,6 +432,27 @@ class Matrix4 {
...
+          _m4storage[1] == 0.0 &&
+          _m4storage[2] == 0.0 &&
+          _m4storage[3] == 0.0 &&
+          _m4storage[4] == 0.0 // col 2
+          &&
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/vector_math.dart:446]: ditto

File: lib/web_ui/lib/src/engine/vector_math.dart [hunk 3]
```diff
@@ -432,6 +432,27 @@ class Matrix4 {
...
+          _m4storage[5] == 1.0 &&
+          _m4storage[6] == 0.0 &&
+          _m4storage[7] == 0.0 &&
+          _m4storage[8] == 0.0 // col 3
+          &&
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/vector_math.dart:451]: ditto

**Review by yjbanov — APPROVED** [review]
lgtm

File: lib/web_ui/lib/src/engine/vector_math.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/vector_math.dart:441]: Done.

File: lib/web_ui/lib/src/engine/vector_math.dart (hunk 2 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/vector_math.dart:446]: Done.

File: lib/web_ui/lib/src/engine/vector_math.dart (hunk 3 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/vector_math.dart:451]: Done.

**Pull request merged**